File: app/src/main/java/com/example/myapplication/atest/RequestException.java
package com.example.myapplication.atest;

import java.io.IOException;

/**
 * @author liuaibin
 */
public class RequestException extends IOException {
    private String mCode;

    public RequestException(String code) {
        super();
        mCode = code;
    }

    public RequestException(String code, String message) {
        super(message);
        mCode = code;
    }

    public String getCode() {
        return mCode;
    }
}

File: app/src/main/java/com/example/myapplication/atestkotlin/RequestException.kt
package com.example.myapplication.atestkotlin

import java.io.IOException

/**
 * Copyright (c) 2022 Raysharp.cn. All rights reserved.
 *
 * RequestException
 * @author liujianming
 * @date 2022-01-05
 */
class RequestException(code: String, message: String) : IOException() {
    private var mCode: String? = null

//    fun RequestException(code: String?) {
//        super@RequestException
//        mCode = code
//    }
//
//    fun RequestException(code: String?, message: String?) {
////        super(message)
//        mCode = code
//    }

    init {
        mCode = code
    }

//    constructor(code: String) : this() {
//        mCode = code
//    }

//    constructor(code: String, message: String) : super(message) {
//    }


    fun getCode(): String? {
        return mCode
    }
}

class Person2(var name: String, var age: Int) {
    init {
        println("main constructor init")
    }

    constructor(name: String, age: Int, sex: String) : this(name, age) {

    }

    constructor(name: String, age: Int, sex: String, idCard: String) : this(name, age, sex) {

    }
}